assets/js/templates/sidebar-items.handlebars

```diff
@@ -1,50 +1,65 @@
-    {{else}}
-      <a href="{{node.id}}.html" title="{{node.title}}" class="expand">{{node.title}}</a>
+      {{! Nodes added purely for nesting purposes (i.e. don't exist in the source) have no id }}
+      {{#if node.id}}
+        {{! When visiting a module page, the link to this module page
+        in the menu should not link to a new page, instead should link
+        to the top of the page itself. }}
+        <a href="{{node.id}}.html" title="{{node.title}}" {{#hasContent node}}class="expand"{{/hasContent}}>
+          {{#if node.title_truncated}}
+            <span class="nested"></span>{{node.title_truncated}}
+          {{else}}
+            {{node.title}}
+          {{/if}}
+        </a>
 
-      <ul>
-        <li>
-          <a href="{{node.id}}.html#content">Top</a>
-        </li>
-
-        {{#isArray node.headers}}
-          {{#each node.headers}}
-            <li>
-              <a href="{{node.id}}.html#{{{anchor}}}">{{id}}</a>
-            </li>
-          {{/each}}
-        {{else}}
-          {{#showSummary node}}
+        {{#hasContent node}}
+          <ul>
             <li>
-              <a href="{{node.id}}.html#summary">Summary</a>
+              <a href="{{node.id}}.html#content">Top</a>
             </li>
-          {{/showSummary}}
-          {{#each node.nodeGroups as |group|}}
-            <li class="docs">
-              <a href="{{node.id}}.html#{{group.key}}" class="expand">{{group.name}}</a>
-              <ul class="{{group.key}}-list deflist">
-                {{#each group.nodes}}
+
+              {{#isArray node.headers}}
+                {{#each node.headers}}
                   <li>
-                    <a href="{{node.id}}.html#{{anchor}}">{{id}}</a>
+                    <a href="{{node.id}}.html#{{{anchor}}}">{{id}}</a>
                   </li>
                 {{/each}}
-              </ul>
-            </li>
-          {{/each}}
-        {{/isArray}}
-      </ul>
-    {{/isEmptyNode}}
-  </li>
-{{/each}}
+              {{else}}
+                {{#showSummary node}}
+                  <li>
+                    <a href="{{node.id}}.html#summary">Summary</a>
+                  </li>
+                {{/showSummary}}
+                {{#each node.nodeGroups as |group|}}
+                  <li class="docs">
+                    <a href="{{node.id}}.html#{{group.key}}" class="expand">{{group.name}}</a>
+                    <ul class="{{group.key}}-list deflist">
+                      {{#each group.nodes}}
+                        <li>
+                          <a href="{{node.id}}.html#{{anchor}}">{{id}}</a>
+                        </li>
+                      {{/each}}
+                    </ul>
+                  </li>
+                {{/each}}
+              {{/isArray}}
+            </ul>
+          {{/hasContent}}
+      {{else}}
+        <span title="{{node.title}}" class="context" aria-hidden="true">
+          {{#if node.title_truncated}}
```

Comment:
This title can never be truncated, right? As we don't support nesting.

Reply:
I believe you're right. I think this is an artifact from the recursive collapse experimentation, but I'll have to check to be sure.